### DISCUSSION TITLE
Controller recognised but not working

### DISCUSSION THREAD

**Discussion by jxrgemartinez** [cathery/sys-con#132] [Troubleshooting]
I am using a controller that can switch between Direct-input and X-input on PC and that is for PlayStation 3.
When it is on DInput, it is not recognised but it appears on the applet that scans devices, but when I switch it to X-input, the switch recognizes it, it doesn't work and in the applet I can't scan, it is like the applet freezes.
Whan can I do?

**Comment by cathery** [comment]
I have no clue. Can you send me the log located at `sdmc:/config/sys-con/log.txt`?

**Comment by jxrgemartinez** [comment]
Herre it is
[log.txt](https://github.com/cathery/sys-con/files/4099307/log.txt)

**Comment by cathery** [comment]
I can't really tell much from the log, but it looks like your controller doesn't implement XInput the same way Microsoft does it. As for DInput I haven't implemented that yet sorry.

**Comment by cathery** [comment]
Also, i see a bunch of config files are missing. Maybe try adding them to your sd card?

**Comment by jxrgemartinez** [comment]
Here's the new log after adding the rest of the config files (they were missing because I tried the solutions for problems similar to mine)
[log.txt](https://github.com/cathery/sys-con/files/4099414/log.txt)

**Comment by jxrgemartinez** [comment]
When I used it on my PC on X-Input, it worked fine (in case it is relevant or something)+

**Comment by jxrgemartinez** [comment]
Hey! I managed to screenshot the results of the applet with my controller, here is the screenshot
The first result is when the controller is on XInput mode, and the second one when it's on DInput
![2020012311555700-DB1426D1DFD034027CECDE9C2DD914B8](https://user-images.githubusercontent.com/60195088/72978902-84976a00-3dd7-11ea-9ed7-fb5e13700c7e.jpg)

**Comment by cathery** [comment]
The XInput appears to have an interface that matches Xbox 360 Controller, but I guess they decided to use HID reports instead (like DInput). I can probably hook it up to use HID reports when available, but you'll have to wait until I add it

**Comment by jxrgemartinez** [comment]
Yeah if I have to wait... I'll wait. But... do you have any idea of how much time could it take you to add it?

**Comment by cathery** [comment]
Whenever it's ready.
